src/test/java/user/TC_UserUpdate.java

```diff
@@ -0,0 +1,133 @@
+import org.testng.asserts.SoftAssert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TC_UserUpdate {
+
+	private final static String NAME = "Kirill";
+
+	private SoftAssert softAssert;
+	private UserDto userDto;
+
+	private List<UserDto> userDtoList;
+
+	@BeforeClass
+	public void initUserListForRemoving() {
+		userDtoList = new ArrayList<>();
+	}
+
+	@BeforeMethod
+	public void createZipCodeAndUser() {
+		softAssert = new SoftAssert();
+		userDto = new UserDto();
+
+		String zipCode = RandomUtils.nextInt(10000, 99999) + "5";
+		ZipCodeService.addZipCodes(Lists.newArrayList(zipCode));
+
+		String uniqueName = NAME + RandomStringUtils.randomAlphabetic(4);
+		userDto.setName(uniqueName);
+		userDto.setAge(RandomUtils.nextInt(1, 99));
+		userDto.setSex(Sex.values()[RandomUtils.nextInt(0, Sex.values().length)]);
+		userDto.setZipCode(zipCode);
+
+		UserService.createUser(userDto);
+	}
+
+	@Test
+	public void verifyUpdateUserTest() {
+		UserDto userUpdateDto = new UserDto();
+		int newAge = RandomUtils.nextInt(1, 99);
+		userUpdateDto.setName(userDto.getName());
+		userUpdateDto.setAge(newAge);
+		userUpdateDto.setSex(userDto.getSex());
+		userUpdateDto.setZipCode(userDto.getZipCode());
+
+		UpdateUserDto updateUser = new UpdateUserDto();
+		updateUser.setUserToChange(userDto);
+		updateUser.setUserNewValues(userUpdateDto);
+
+		int statusCode = UserService.updateUser(updateUser);
+		userDtoList.add(userUpdateDto);
+		softAssert.assertEquals(statusCode, HttpStatus.SC_OK,
+				"Response code is NOT 200 when updating a user.");
+
+		int actualAge = UserService.findUsersByName(userUpdateDto.getName()).get(0).getAge();
+		softAssert.assertEquals(actualAge, newAge, "User is NOT updated.");
+		softAssert.assertAll();
+	}
+
+	@Test
+	public void verifyUpdateUserWithIncorrectZipCodeTest() {
+		UserDto userUpdateDto = new UserDto();
+		int newAge = RandomUtils.nextInt(1, 99);
+		String incorrectZipCode = RandomUtils.nextInt(10000, 99999) + "61";
+		userUpdateDto.setName(userDto.getName());
+		userUpdateDto.setAge(userDto.getAge());
+		userUpdateDto.setSex(userDto.getSex());
+		userUpdateDto.setZipCode(incorrectZipCode);
+
+		UpdateUserDto updateUser = new UpdateUserDto();
+		updateUser.setUserToChange(userDto);
+		updateUser.setUserNewValues(userUpdateDto);
+
+		int statusCode = UserService.updateUser(updateUser);
+		userDtoList.add(userUpdateDto);
+		softAssert.assertEquals(statusCode, HttpStatus.SC_FAILED_DEPENDENCY,
+				"Response code is NOT 424 when updating a user with incorrect zip code.");
+
+		List<UserDto> users = UserService.findUsersByName(userUpdateDto.getName());
+		softAssert.assertFalse(users.isEmpty(), "Updated user does not exist in users.");
+
+		softAssert.assertAll();
+
+		int actualAge = users.get(0).getAge();
+		Assert.assertEquals(actualAge, newAge, "User is updated when updating a user with incorrect zip code.");
+	}
+
+	@Test
+	public void verifyUpdateUserWithRequiredFieldsMissedTest() {
```

Comment:
This test verifies update operation for userNewValues without required fields. Please add one more test when userToChange doesn't have required fields

Reply:
Added one more test where 'userToChange' doesn't have required fields.